ccmlib/scylla_repository.py

```diff
@@ -234,15 +234,47 @@ def setup(version, verbose=True, skip_downloads=False):
               - release:4.2.1
               - release:2020.1
               - release:2020.1.5
+            flags can be added to the version using [(flag(=value)?)+]
+              - supported flags are:
+                - product=(scylla|scylla-enterprise)
+                - arch=(x86_64|aarch64)
+                - debug
 
     :param verbose: if True, print progress during download
     :param skip_downloads: if True, skips the actual download of files for testing purposes
 
     """
     s3_url = ''
-    type_n_version = version.split(':', 1)
-    scylla_product = os.environ.get('SCYLLA_PRODUCT', 'scylla')
-    scylla_arch = os.environ.get('SCYLLA_ARCH', 'x86_64')
+    version_re = re.compile(r"""
+                            (?P<type>
+                                \w+(?:/(?:\w+))?
+                            )
+                            :(?P<version>[\w\d.:-]+)
+                            (?:
+                                \[
+                                    (?P<flags>[^\]]*)
+                                \]
+                            )?""", re.VERBOSE)
+    m = version_re.match(version)
+    assert m, f"Invalid version {version}: use (release:version|unstable/branch:(latest|timestamp))([(flag(=value)?)*])? where flags can be any of: (product|arch|debug)"
+    type_n_version = [m.group("type"), m.group("version")]
+    flags = dict()
+    if m.group("flags"):
+        for kv in m.group("flags").split(","):
+            t = kv.split("=", 1)
+            key = t[0].strip().lower()
+            if len(t) > 1:
+                value = t[1].strip()
+                if value.lower() in ["true", "yes", "1"]:
```

Comment:
I don't understand this part, non of those flags has boolean content, why do we need this ?

Reply:
It's a little semantic sugar if someone wants to specify `[debug=true]` instead of `[debug]`
since the syntax allows `[debug=value]`

Reply:
it's not a must-have by any means

Reply:
removed in v2